R/read_table.R

```diff
@@ -17,16 +17,9 @@
 #' @inheritParams read_delim
 #' @export
 #' @examples
-#' # One corner from http://www.masseyratings.com/cf/compare.htm
-#' massey <- readr_example("massey-rating.txt")
-#' cat(read_file(massey))
-#' read_table(massey)
-#'
-#' # Sample of 1978 fuel economy data from
-#' # http://www.fueleconomy.gov/feg/epadata/78data.zip
-#' epa <- readr_example("epa78.txt")
-#' cat(read_file(epa))
-#' read_table(epa, col_names = FALSE)
+#' ws <- readr_example("whitespace-sample.txt")
+#' writeLines(read_file(ws))
+#' read_table(ws, col_names = FALSE)
```

Comment:
```suggestion
#' writeLines(read_lines(ws))
#' read_table(ws)
```

I think we absolutely want the `col_names` in this case. Also making the switch to `read_lines()`, as explained above.

``` r
devtools::load_all()
#> ℹ Loading readr

ws <- readr_example("whitespace-sample.txt")
writeLines(read_lines(ws))
#> first last state phone
#> John Smith WA 418-Y11-4111
#> Mary Hartford CA 319-Z19-4341
#> Evan Nolan IL 219-532-c301

# this produces incorrect column names and incorrect data
read_table(ws, col_names = FALSE)
#> 
#> ── Column specification ────────────────────────────────────────────────────────
#> cols(
#>   X1 = col_character(),
#>   X2 = col_character(),
#>   X3 = col_character(),
#>   X4 = col_character()
#> )
#> # A tibble: 4 × 4
#>   X1    X2       X3    X4          
#>   <chr> <chr>    <chr> <chr>       
#> 1 first last     state phone       
#> 2 John  Smith    WA    418-Y11-4111
#> 3 Mary  Hartford CA    319-Z19-4341
#> 4 Evan  Nolan    IL    219-532-c301

# this produces correct column names and correct data
read_table(ws)
#> 
#> ── Column specification ────────────────────────────────────────────────────────
#> cols(
#>   first = col_character(),
#>   last = col_character(),
#>   state = col_character(),
#>   phone = col_character()
#> )
#> # A tibble: 3 × 4
#>   first last     state phone       
#>   <chr> <chr>    <chr> <chr>       
#> 1 John  Smith    WA    418-Y11-4111
#> 2 Mary  Hartford CA    319-Z19-4341
#> 3 Evan  Nolan    IL    219-532-c301
```

<sup>Created on 2022-01-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1.9000)</sup>